CMakeLists.txt

```diff
@@ -51,6 +51,14 @@ endif()
 set(NVIM_VERSION_MAJOR 0)
 set(NVIM_VERSION_MINOR 1)
 set(NVIM_VERSION_PATCH 1)
+# Package maintainers should define this according to their package system
+# conventions, if the source code or runtime files were modified, e.g.:
+#   cmake -DNVIM_VERSION_PRERELEASE=-deb3
+# or define it as empty to prevent "-dev", if no modifications were made:
+#   cmake -DNVIM_VERSION_PRERELEASE=
+if(NOT DEFINED NVIM_VERSION_PRERELEASE)
+  set(NVIM_VERSION_PRERELEASE "-dev")
+endif()
```

Comment:
I don't like this too much.. I think the version should not depend on how Nvim is built. If a source snapshot is not a release, it should have "-dev", otherwise it shouldn't. I agree that it's cumbersome to remove `NVIM_VERSION_PRERELEASE` when we tag a release, though.. but I don't have a better idea.

If a packager modifies code or runtime files via patches, they can as well do that with the CMake code, so we don't need to expose a CMake property for this use-case.

Semi-related: currently we number versions like: `0.1.0 -> 0.1.0-dev -> 0.1.1 -> 0.1.1-dev -> ...` This doesn't follow SemVer:

> When major, minor, and patch are equal, a pre-release version has lower precedence than a normal version. Example: 1.0.0-alpha < 1.0.0.

If we number our versions like `0.1.0 -> 0.1.1-dev -> 0.1.1 -> 0.1.2-dev -> ...`, we anyway have to add a follow-up commit after tagging a release, to increase the version number. This commit could also re-introduce a hardcoded `NVIM_VERSION_PRERELEASE`.

I didn't know how other projects do this, so I picked `libuv` more or less at random.. looks like they do what I described: [tagging a release](https://github.com/libuv/libuv/commit/5467299450ecf61635657557b6e01aaaf6c3fdf4#diff-6da6ac057a6e89d11bde410b064ec16eL34), [follow-up commit](https://github.com/libuv/libuv/commit/61f13a6d5e52817c748ddedea73efcf3eb94c089)

Now that I think of it, what I described so far only applies to "building from tarballs"... `git describe` complicates the situation somewhat. We don't really follow SemVer anymore since we introduced that. Hm.